server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/JMAPApiRoutes.scala

```diff
@@ -101,14 +111,12 @@ class JMAPApiRoutes @Inject() (serializer: Serializer) extends JMAPRoutes {
       invocation.methodCallId))
   }
 
-  private def handleError(throwable: Throwable, httpServerResponse: HttpServerResponse): SMono[Void] = {
-    if (throwable.isInstanceOf[IllegalArgumentException]) {
-      return SMono.fromPublisher(httpServerResponse.status(SC_BAD_REQUEST)
-        .header(CONTENT_TYPE, JSON_CONTENT_TYPE)
-        .sendString(SMono.fromCallable(() => throwable.getMessage), StandardCharsets.UTF_8)
-        .`then`())
-    }
-
-    SMono.fromPublisher(handleInternalError(httpServerResponse, throwable))
+  private def handleError(throwable: Throwable, httpServerResponse: HttpServerResponse): SMono[Void] = throwable match {
+    case exception: IllegalArgumentException => SMono.fromPublisher(httpServerResponse.status(SC_BAD_REQUEST)
+      .header(CONTENT_TYPE, JSON_CONTENT_TYPE)
+      .sendString(SMono.fromCallable(() => exception.getMessage), StandardCharsets.UTF_8)
```

Comment:
I disagree with this, we should not leak the exception to the end user, it might be sensible.